src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IAsyncEnumerableConverterFactory.cs

```diff
@@ -15,12 +15,15 @@ namespace System.Text.Json.Serialization
     internal sealed class IAsyncEnumerableConverterFactory : JsonConverterFactory
     {
         [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+        [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
         public IAsyncEnumerableConverterFactory() { }
 
         public override bool CanConvert(Type typeToConvert) => GetAsyncEnumerableInterface(typeToConvert) is not null;
 
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
             Justification = "The ctor is marked RequiresUnreferencedCode.")]
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
+            Justification = "The ctor is marked with RequiresDynamicCode.")]
```

Comment:
```suggestion
            Justification = "The ctor is marked RequiresDynamicCode.")]
```
Nit: Make the ReflectionAnalysis and AotAnalysis justifications look the same.

Reply:
(multiple places)